src/main.c

```diff
@@ -81,40 +83,107 @@ static u8_t no_exit = 0;
+    printf("\t--noexit, -n               Do not exit jslinux when no events"
+                                        " are present\n");
+    printf("\t--unittest, -u             Run jslinux unit tests\n");
+    printf("\t--debugger, -d             Start debugger, waits for debugging"
+                                        " connection at startup\n");
+    printf("\t--help, -h                 Display this help message\n");
+}
+
+static const struct option main_options[] = {
+    { "jsargs", required_argument, NULL, 'j' },
+    { "exit-time",	required_argument, NULL, 't' },
+    { "noexit",     no_argument, NULL, 'n' },
+    { "unittest",   no_argument, NULL, 'u' },
+    { "debugger",	no_argument, NULL, 'd' },
+    { "help",       no_argument, NULL, 'h' },
+    { }
+};
 
 u8_t process_cmd_line(int argc, char *argv[])
 {
-    int i;
-    for (i = 0; i < argc; ++i) {
-        if (!strncmp(argv[i], "--unittest", 10)) {
-            // run unit tests
+    for (;;) {
+        int opt;
+
+        opt = getopt_long(argc, argv, "j:t:nudh", main_options, NULL);
+        if (opt < 0)
+            break;
+
+        switch (opt) {
+        case 't':
+            exit_after = atoi(optarg);
+            ZJS_PRINT("jslinux will terminate after %u milliseconds\n",
+                          (unsigned int)exit_after);
+            clock_gettime(CLOCK_MONOTONIC, &exit_timer);
+            break;
+        case 'n':
+            no_exit = true;
+            break;
+        case 'u':
             zjs_run_unit_tests();
-        } else if (!strncmp(argv[i], "--debugger", 10)) {
+            break;
+        case 'd':
 #ifdef ZJS_DEBUGGER
             // run in debugger
             start_debug_server = true;
 #else
             ERR_PRINT("Debugger disabled, rebuild with DEBUGGER=on");
             return 0;
 #endif
-        } else if (!strncmp(argv[i], "--noexit", 8)) {
-            no_exit = 1;
-        } else if (!strncmp(argv[i], "-t", 2)) {
-            if (i == argc - 1) {
-                // no time argument, return error
-                ERR_PRINT("no time argument given after '-t'\n");
-                return 0;
-            } else {
-                char *str_time = argv[i + 1];
-                exit_after = atoi(str_time);
-                ZJS_PRINT("jslinux will terminate after %u milliseconds\n",
-                          (unsigned int)exit_after);
-                clock_gettime(CLOCK_MONOTONIC, &exit_timer);
-            }
+        case 'h':
+            usage();
+            exit(0);
+        case 'j':
+            js_args = optarg;
+            break;
+        default:
+            usage();
+            exit(1);
         }
     }
+
     return 1;
 }
+
+void init_process(char *args)
+{
+    if (!args)
+        return;
+
+    char *ptr = args;
+    char *last = args;
+    jerry_value_t array = ZJS_UNDEFINED;
+    ZVAL global_obj = jerry_get_global_object();
+    ZVAL process = zjs_create_object();
+
+    while ((ptr = strchr(last, ' '))) {
```

Comment:
I'd prefer you do the assignment outside of the condition. So probably make it a forever loop that you break out of.